File: internal/core/plan.go
package core

import (
	"time"

	"github.com/gofrs/uuid/v5"
)

type Plan struct {
	ID            uuid.UUID `db:"id"`
	Name          string    `db:"name"`
	Price         int       `db:"price"`
	StripePriceID string    `db:"stripe_price_id"`
	CreatedAt     time.Time `db:"created_at"`
	UpdatedAt     time.Time `db:"updated_at"`
}

File: frontend/src/api/modules/plan.ts
import * as api from '@/api/instance';

export type Plan = {
  id: string;
  name: string;
  price: number;
  createdAt: string;
  updatedAt: string;
};

export const getPlans = async () => {
  const res = await api.get<{
    plans: Array<Plan>;
  }>({
    path: '/plans',
    auth: true,
  });

  return res;
};
